xaitk_saliency/impls/saliency/sliding_window.py

```diff
@@ -0,0 +1,120 @@
+from typing import Tuple, List
+
+from xaitk_saliency import PerturbImage
+from xaitk_saliency.utils.masking import generate_masked_images
+
+import numpy as np
+import PIL.Image
+
+
+class SlidingWindow(PerturbImage):
+    """
+    Implementation of a sliding window based perturbation taking a reference
+    image and generating some number perturbations of the image along with
+    paired mask matrices indicating where perturbations have occurred and
+    to what amount.
+
+    The resolution of perturbation is dependant upon the window size and
+    stride of the sliding window.
+    """
+
+    def __init__(self,
+                 window_size: int,
+                 stride: int) -> None:
+        self.window_size = window_size
+        self.stride = stride
+
+    def get_config(self) -> dict:
+        return {"window_size": self.window_size,
+                "stride": self.stride}
+
+    def generate_block_masks(
+        self,
+        window_size: int,
+        stride: int,
+        image_size: Tuple[int, int]
+    ) -> np.ndarray:
+        """
+        Generates sliding window type binary masks used in augment() to
+        mask an image.
+        :param window_size: the block window size (with value 0, other areas
+            with value 1)
+        :param stride: the sliding step
+        :param image_size: the mask size which should be the same to the image size
+
+        :return: the sliding window style masks with values ranging between [0, 1].
+        """
+        rows = np.arange(0 + stride - window_size, image_size[0], stride)
+        cols = np.arange(0 + stride - window_size, image_size[1], stride)
+
+        mask_num = len(rows) * len(cols)
+        masks = np.ones((mask_num, image_size[0], image_size[1]),
+                        dtype=np.float64)
```

Comment:
dtype can probably be boolean or uint8 if we care about saving memory/storage since the masks here are binary.

Reply:
Done. Changed it to use uint8 masks

Reply:
This is subsumed by the base branch (which uses bool mask generation).